ts/screens/main/MessagesScreen.tsx

```diff
@@ -1,24 +1,85 @@
-import { Container } from "native-base";
+import {
+  Body,
+  Container,
+  Content,
+  H1,
+  Tab,
+  Tabs,
+  Text,
+  View
+} from "native-base";
 import * as React from "react";
-import { NavigationScreenProp, NavigationState } from "react-navigation";
+import { connect } from "react-redux";
+import { CreatedMessageWithContent } from "../../../definitions/backend/CreatedMessageWithContent";
+import { ReduxProps } from "../../actions/types";
+import messages from "../../api/mock/messages.json";
+import MessageComponent from "../../components/MessageComponent";
+import AppHeader from "../../components/ui/AppHeader";
+import I18n from "../../i18n";
+import variables from "../../theme/variables";
 
-export type OwnProps = {
-  navigation: NavigationScreenProp<NavigationState>;
+type ReduxMappedProps = {};
+export type OwnProps = {};
+type Props = ReduxMappedProps & ReduxProps & OwnProps;
+type State = {};
+type ListOfCreatedMessageWithContentInterface = {
+  [id: string]: CreatedMessageWithContent;
 };
 
-export type Props = OwnProps;
-
 /**
  * This screen show the messages to the authenticated user.
  */
-class MessagesScreen extends React.Component<Props, never> {
-  constructor(props: Props) {
-    super(props);
-  }
+class MessagesScreen extends React.Component<Props, State> {
+  public renderMessageList = (
+    list: ListOfCreatedMessageWithContentInterface
+  ) => {
+    return Object.keys(list).map(key => {
```

Comment:
`Object.keys` will return an unordered list of messages - instead, messages should be sorted by ID. We may want to keep a sorted list of message IDs in the state to avoid sorting it each time.

Reply:
I have suppose that the store return an ordred list of messages. The ordination may be after normalaized the message.